docs/proposals/network-policy.md

```diff
@@ -0,0 +1,136 @@
+# Native API Support for Network Policy
+#### Abstract
+A proposal for adding the API primitives to describe flexible network policy.
+
+#### Goals
+- Add first-class support for flexible network policy to the API.
+- Provide an intent-based API which uses native Kubernetes concepts (labels, namespaces, services).
+- Abstracts policy from the underlying networking technology.
+- Back-compatible with existing network plugins.  Network plugins can choose which (if any) of this API is supported.
+
+#### Proposed Design
+The design uses Kubernetes labels to apply policy to pods.  A NetworkPolicy object is introduced which uses a label
+selector to apply a list of intent-based rules to groups of pods.  If traffic matches any rule within the given list, it
+should be allowed to the destination pod.  If no rules are matched, it should be rejected.
+
+Each rule in the NetworkPolicy can apply policy on any combination of the following fields:
+- Kubernetes Namespace
+- Pod Labels
+- Protocol (TCP vs UDP)
+- Target port
+
+The rules can be extended to support additional fields in the future.
+
+##### New API Objects
+I'd like to propose the following set of new first-class API objects
```

Comment:
Make this all third-person, since this document is a spec (and could be edited/updated by multiple authors).